src/util/chunk_fs.js

```diff
@@ -28,54 +28,50 @@ class ChunkFS extends stream.Transform {
+            });
+            this.count = 0;
+            while (chunk && chunk.length) {
+                const available_size = config.NSFS_BUF_SIZE - this.q_size;
+                const buf = (available_size < chunk.length) ? chunk.slice(0, available_size) : chunk;
+                this.q_buffers.push(buf);
+                this.q_size += buf.length;
+                if (this.q_size === config.NSFS_BUF_SIZE) await this._flush_buffers();
+                chunk = (available_size < chunk.length) ? chunk.slice(available_size) : null;
+            }
+            return callback();
         } catch (error) {
             return callback(error);
         }
     }
 
-    _flush(callback) {
-        try {
-            this._flush_buffers(callback);
-        } catch (error) {
-            return callback(error);
-        }
+    async _flush(callback) {
+        this._flush_buffers(callback);
     }
 
     async _flush_buffers(callback) {
-        if (this.q_buffers.length) {
-            const buffers_to_write = this.q_buffers;
-            this.q_buffers = [];
-            this.q_size = 0;
-            await this.target_file.writev(this.fs_account_config, buffers_to_write);
-            // Hold the ref on the buffers from the JS side
-            this._total_num_buffers += buffers_to_write.length;
-        }
-        if (callback) {
-            if (this.MD5Async) this.digest = (await this.MD5Async.digest()).toString('hex');
-            return callback();
-        }
-    }
-
-    async _process_chunk(data, callback) {
-        if (this.MD5Async) await this.MD5Async.update(data);
-        stats_collector.instance(this.rpc_client).update_namespace_write_stats({
-            namespace_resource_id: this.namespace_resource_id,
-            size: data.length,
-            count: this.count
-        });
-        this.count = 0;
-        while (data && data.length) {
-            const available_size = config.NSFS_BUF_SIZE - this.q_size;
-            const buf = (available_size < data.length) ? data.slice(0, available_size) : data;
-            this.q_buffers.push(buf);
-            this.q_size += buf.length;
-            if (this.q_size === config.NSFS_BUF_SIZE) await this._flush_buffers();
-            data = (available_size < data.length) ? data.slice(available_size) : null;
+        try {
+            if (this.q_buffers.length) {
+                const buffers_to_write = this.q_buffers;
+                this.q_buffers = [];
+                this.q_size = 0;
+                await this.target_file.writev(this.fs_account_config, buffers_to_write);
+                // Hold the ref on the buffers from the JS side
+                this._total_num_buffers += buffers_to_write.length;
+            }
+            if (callback) {
+                if (this.MD5Async) this.digest = (await this.MD5Async.digest()).toString('hex');
+                return callback();
+            }
+        } catch (error) {
+            return callback(error);
```

Comment:
Perhaps this should be printed here as warning/error? sounds like this is wouldn't be a good thing that we fail during this process of flushing data from memory to file... same for transform..
```suggestion
            console.error('ChunkFS FLUSH TO FILE FAILED', this.q_size, this._total_num_buffers, error);
            return callback(error);
```